tools/prow-config-updater/updater.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"knative.dev/pkg/test/cmd"
+	"knative.dev/pkg/test/ghutil"
+	"knative.dev/pkg/test/helpers"
+
+	"knative.dev/test-infra/shared/common"
+	"knative.dev/test-infra/tools/prow-config-updater/config"
+)
+
+type ConfigUpdater struct {
+	githubmainhandler *GitHubMainHandler
+	githubcommenter   *GitHubCommenter
+	prnumber          int
+	files             []string
+	dryrun            bool
+}
+
+const (
+	// This command can be used for both production and staging prow (%s can be replace as the environment name).
+	updateProwCommandTemplate = "make -C %s update-prow-cluster"
+)
+
+var (
+	// These two commands are only used for production prow in this tool.
+	updateTestgridCommand        = fmt.Sprintf("make -C %s update-testgrid-config", config.ProdProwConfigRoot)
+	updateProwConfigFilesCommand = fmt.Sprintf("make -C %s config", config.ProdProwConfigRoot)
+)
+
+func runProwConfigUpdate(mgc *ghutil.GithubClient, cgc *ghutil.GithubClient, dryrun bool) error {
+	githubmainhandler := &GitHubMainHandler{client: mgc, dryrun: dryrun}
+	pr, err := githubmainhandler.getLatestPullRequest()
+	if err != nil {
+		return fmt.Errorf("error getting the latest PR number: %v", err)
+	}
+	fs, err := githubmainhandler.getChangedFiles(*pr.Number)
+	if err != nil {
+		return fmt.Errorf("error getting changed files in PR %q: %v", *pr.Number, err)
+	}
+
+	if err := common.CDToRootDir(); err != nil {
+		return err
+	}
+
+	cu := &ConfigUpdater{
+		githubmainhandler: githubmainhandler,
+		githubcommenter:   &GitHubCommenter{client: cgc, dryrun: dryrun},
+		prnumber:          *pr.Number,
+		files:             fs,
+		dryrun:            dryrun,
+	}
+	// If the PR is created by the Prow bot, we should be confident to blindly update production Prow configs.
+	if *pr.User.Name == config.ProwBotName {
+		if err := cu.updateProw(config.ProdProwEnv); err != nil {
```

Comment:
this could be deduped from # 75

Reply:
I have modified the logic to reduce the duplication a bit.